app/controllers/events_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :log_in_user, only: [:create]
+
+  def new
+    if signed_in?
+      @event = current_user.events.build
+    else
+      flash[:danger] = 'Kindly log in to create an event'
+      redirect_to signin_path
+    end
+  end
+
+  def create
+    @event = current_user.events.build(event_params)
+    if @event.save
+      redirect_to event_path(id: @event.id)
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @events = Event.all
+    @past_events = @events.past
+    @upcoming_events = @events.upcoming
```

Comment:
`[Optional]` Here you could use `@past_events = Event.past`, the same for `@upcoming_events` and in the `index` view you could save the `if @events.any?` validations because you are using the same validation inside the `upcoming_past` partial.